gravitee-apim-gateway/gravitee-apim-gateway-reactor/src/main/java/io/gravitee/gateway/jupiter/reactor/DefaultHttpRequestDispatcher.java

```diff
@@ -157,23 +158,23 @@ public Completable dispatch(HttpServerRequest httpServerRequest) {
                     mutableCtx,
                     ExecutionPhase.REQUEST
                 )
-                .andThen(
-                    Completable.defer(
-                        () -> {
-                            // Jupiter execution mode.
-                            ProcessorChain postProcessorChain = platformProcessorChainFactory.postProcessorChain();
-                            return handleJupiterRequest(mutableCtx, httpAcceptor)
-                                .andThen(
-                                    HookHelper.hook(
-                                        () -> postProcessorChain.execute(mutableCtx, ExecutionPhase.RESPONSE),
-                                        postProcessorChain.getId(),
-                                        processorChainHooks,
-                                        mutableCtx,
-                                        ExecutionPhase.RESPONSE
-                                    )
-                                );
-                        }
-                    )
+                .andThen(Completable.defer(() -> handleJupiterRequest(mutableCtx, httpAcceptor)))
+                .doFinally(
+                    () -> {
+                        ProcessorChain postProcessorChain = platformProcessorChainFactory.postProcessorChain();
+
+                        HookHelper
+                            .hook(
+                                () -> postProcessorChain.execute(mutableCtx, ExecutionPhase.RESPONSE),
+                                postProcessorChain.getId(),
+                                processorChainHooks,
+                                mutableCtx,
+                                ExecutionPhase.RESPONSE
+                            )
+                            .subscribeOn(Schedulers.computation())
+                            .onErrorComplete()
+                            .subscribe();
```

Comment:
Not fan of this but I guess you are doing it to make the post platform processor independant from the response. Could you add some log here?

Reply:
Exactly! Previously we were missing some precious logs (ex: client cancellation) so putting it on the doFinally allows to a double benefit by dissociating the post actions from the event loop and cover all situations.
I'll add a comment on that 👍